lib/helper.js

```diff
@@ -984,6 +993,61 @@ helper = {
     });
 
     return null;
+  },
+
+  /**
+   * Resolves a resource by inheriting values from global resourceTypes as defined for corresponding resource
+   *
+   * @param {*} resource - Resource to be resolved
+   * @param {*} resourceTypes - Resource types defined at global level of RAML definition
+   * @returns {*} - Resolved resource
+   */
+  resolveResource: function (resource, resourceTypes) {
+    let resolvedResource = resource,
+      inheritedResource;
+
+    if (resource && !_.isEmpty(resource.type)) {
+      inheritedResource = _.find(resourceTypes, (resourceType) => {
```

Comment:
For each `resource`, we iterate over `resourceTypes`. This can be optimised by having `resourceTypeMap` keyed by `resource.type` to we avoid unnecessary iterations.